internal/plugin/connectors/http/generic/oauth/v1/protocol.go

```diff
@@ -0,0 +1,275 @@
+	paramMap[oauthToken] = []string{encodeURI(oauth1.Token)}
+	paramMap[oauthVersion] = []string{encodeURI(oauth1.Version)}
+
+	return paramMap
+}
+
+/*
+generateParameterString creates a Parameter String from sorting an encoded map of parameters and values
+
+Reference: https://tools.ietf.org/html/rfc5849#section-3.5
+*/
+func generateParameterString(paramMap map[string][]string) string {
+	var parameterString bytes.Buffer
+
+	// make map of keys and sort to append lexicographically later
+	keys := make([]string, 0, len(paramMap))
+	for k := range paramMap {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	// create string
+	for keyIndex, key := range keys {
+		// Write keys and value pairs to `parameterString`
+		// ex: key0=value&key1=value&key2=value
+		for valueIndex, value := range paramMap[key] {
+			// Do not add '&' on first value
+			if keyIndex != 0 || valueIndex > 0 {
+				parameterString.WriteString("&")
+			}
+			parameterString.WriteString(key + "=" + value)
+		}
+	}
+
+	return parameterString.String()
+}
+
+/*
+generateBaseString combines parameters to create the pre-hash base string for the signature
+
+Reference: https://tools.ietf.org/html/rfc5849#section-3.4.1
+*/
+func generateBaseString(r *gohttp.Request, paramString string) string {
+	method := strings.ToUpper(r.Method)
+	urlString := fmt.Sprintf("%s://%s%s", r.URL.Scheme, r.URL.Host, r.URL.Path)
+	return fmt.Sprintf("%s&%s&%s", method, encodeURI(urlString), encodeURI(paramString))
+}
+
+/*
+generateSigningKey concats the consumer secret and token secret with an ampersand, and encodes them
+Reference: https://tools.ietf.org/html/rfc5849#section-3.4
+*/
+func generateSigningKey(oauth1 oAuth1) string {
+	return fmt.Sprintf("%s&%s", encodeURI(oauth1.ConsumerSecret), encodeURI(oauth1.TokenSecret))
+}
+
+/*
+hashSignature creates a HMAC-SHA1 hash
+
+Reference: https://tools.ietf.org/html/rfc5849#section-3.4.2
+*/
+func hashSignature(signingKey string, baseString string) string {
+	h := hmac.New(sha1.New, []byte(signingKey))
+	h.Write([]byte(baseString))
+	return encodeURI(base64.StdEncoding.EncodeToString(h.Sum(nil)))
+}
+
+/*
+constructOAuthString creates the Authorization header that is needed for the final request
+
+Reference: https://tools.ietf.org/html/rfc5849#section-3.5.1
+*/
+func constructOAuthString(oauth1 oAuth1) (string, error) {
+	tmpl, err := template.New("oauth").Parse("OAuth " +
+		"oauth_consumer_key=\"{{.ConsumerKey}}\", " +
+		"oauth_nonce=\"{{.Nonce}}\", " +
+		"oauth_signature=\"{{.Signature}}\", " +
+		"oauth_signature_method=\"{{.SignatureMethod}}\", " +
+		"oauth_timestamp=\"{{.TimeStamp}}\", " +
+		"oauth_token=\"{{.Token}}\", " +
+		"oauth_version=\"{{.Version}}\"")
+	if err != nil {
+		return "", fmt.Errorf("could not create the OAuth template")
+	}
+	var tpl bytes.Buffer
+	err = tmpl.Execute(&tpl, oauth1)
+	if err != nil {
+		return "", fmt.Errorf("could not turn the OAuth template into a string")
+	}
+	return tpl.String(), nil
+}
```

Comment:
Lots of squished code here making it hard to read. Add newlines after each `if err != nil {}` block.